pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImplementationBindingImpl.java

```diff
@@ -252,7 +252,11 @@ public <K, V> Schema<KeyValue<K, V>> newKeyValueSchema(Class<K> key, Class<V> va
     }
 
     public Schema<?> getSchema(SchemaInfo schemaInfo) {
-        return AutoConsumeSchema.getSchema(schemaInfo);
+        if (schemaInfo.getType() == SchemaType.PROTOBUF) {
```

Comment:
if change this method is, it is a breaking change. I suggest adding a new method in `AutoProduceBytesSchema`